cdc/log.cc

```diff
@@ -630,7 +630,7 @@ class transformer final {
             const mutation& m)
     {
         auto& p = m.partition();
-        if (p.partition_tombstone() || !p.row_tombstones().empty() || p.clustered_rows().empty()) {
+        if (p.clustered_rows().empty() && p.static_row().empty()) {
```

Comment:
I don't understand this change. A better commit message would have helped to understand what you are fixing here.

Reply:
Commit message will be added but in the meantime, I will explain.

We want to generate preimage any time there's a clustered or static row in the mutation.
The mutation can contain multiple operations: partition deletion, range tombstone, insert/update/delete in clustering or static row
Any operation on clustering or static row requires preimage. Other operations (partition deletion and range tombstone) have empty preimage.
That's why we want to obtain preimage when there's at least one clustering or static row in the mutation. It means there's operation on clustering or static row and we need preimage for it.

Please let me know if I failed to explain it, @nyh